c11threads.h

```diff
@@ -60,25 +123,164 @@ enum {
 	thrd_nomem
 };
 
+/* Global data functions. */
+
+/* Win32: Initialize global data structures. */
+static inline void thrd_init(void);
+/* Win32: Destroy global data structures. */
+static inline void thrd_destroy(void);
```

Comment:
Are these (`thrd_init`/`thrd_destroy`/`thrd_self_register`) in the standard? It's been a while since I've read it. If not, then isn't there a way to avoid them, like lazy initialization on first use?

If they can't be avoided, they should have distinct prefixes from the standard functions. Maybe `c11threads_` to make it obvious this is just an implementation detail of this library and not a standard function which can be used on portable programs.

Reply:
These aren't standard and I'll rename them. I was going with lazy initialization first, but when I noticed we'd have to free some resources anyway with some kind of dtor function figured I'd rather have the user initialize once and not bother with lazy initialization. I think that wouldn't be the right way because it could lead to delays. In a dynamic library, this would be done on load, and in the C library, it would be done on startup, so I decided to mirror that as best as possible.

Reply:
I'd rather not require the user to call any extra functions to use this, other than what they'd normally use if their standard library had native c11 threads support.

Is there a reason why we can't let those data structures be for the whole lifetime of the process and skip the destructor?

Reply:
A possible scenario could be that we're inside a library that needs to be unloaded.

Reply:
True, but that's a corner case. Let's keep the destroy function then (and the init for symmetry), but make init optional by lazy-initializing when necessary, and leave it to the user if they need to bother with destroy or not. But do not make these prominent. Put the win32 implementation-specific functions after the declarations of all the standard functions in the header file.

Reply:
Will do.